kratos/processes/check_same_modelpart_using_skin_distance_process.h

```diff
@@ -72,7 +72,26 @@ class KRATOS_API(KRATOS_CORE) CheckSameModelPartUsingSkinDistanceProcess
         Model& rModel,
         Parameters ThisParameters = Parameters(R"({})")
         )
-        : mrModel(rModel),
+        : mrSkinModelPart1(rModel.GetModelPart(ThisParameters["skin_model_part_1_name"].GetString())),
+          mrSkinModelPart2(rModel.GetModelPart(ThisParameters["skin_model_part_2_name"].GetString())),
+          mThisParameters(ThisParameters)
+    {
+        KRATOS_TRY
+
+        Parameters default_parameters = GetDefaultParameters();
+        mThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+        KRATOS_CATCH("");
+    }
+
+    /// Default constructor.
+    explicit CheckSameModelPartUsingSkinDistanceProcess(
+        ModelPart& rSkinModelPart1,
+        ModelPart& rSkinModelPart2,
+        Parameters ThisParameters = Parameters(R"({})")
+        )
+        : mrSkinModelPart1(rSkinModelPart1),
+          mrSkinModelPart2(rSkinModelPart2),
           mThisParameters(ThisParameters)
```

Comment:
Why adding a pure model part constructor? The line of the project is to promote the model-based ones. If the reasoning behind is to use more than one model, in my opinion giving this possibility is not a good idea. Note that the line of the project is to have a unique `Model` container, also in the multistage case.

Reply:
(yes, that's the idea, @pooyan-dadvand wants to use it to check it in CoSim and the same model is not guaranteed)

Reply:
Though this is a fair usage, I'm not sure if we want to open this possibility in the entire Kratos core. To be discussed in @KratosMultiphysics/technical-committee.